### PR TITLE
Make reloading client certificate configuration more reliable

### LINKED ISSUE #7312: Client cert reload from sni.yaml not reliable
@thebadpete had problems with "traffic_ctl config reload" not picking up updates of client certificate files specified in sni.yaml even after touching the sni.yaml file.  This case is exercised in the tls_client_cert autest, but with the larger sni.yaml file used in production, the sni.yaml was being reloaded but new client certificates were not taking affect without traffic_server restart.

### PR DESCRIPTION
**Pull request by shinrich**
Addresses configuration reload inconsistencies by adding a coordinating update class to ensure that the SSLConfig which owns the client cert storage is always reloaded before the SNIConfig which processes the sni.yaml file.

This patch addressed the issue seen by @thebadpete.  This fix is also part of the larger cert reloading PR #6609

This closes #7312

### CODE REVIEW

File: iocore/net/SSLConfig.cc [hunk 1]
```diff
@@ -426,20 +427,31 @@ SSLConfigParams::getClientSSL_CTX() const
...
+
 void
 SSLConfig::startup()
 {
-  sslConfigUpdate.reset(new ConfigUpdateHandler<SSLConfig>());
-  sslConfigUpdate->attach("proxy.config.ssl.client.cert.path");
-  sslConfigUpdate->attach("proxy.config.ssl.client.cert.filename");
-  sslConfigUpdate->attach("proxy.config.ssl.client.private_key.path");
-  sslConfigUpdate->attach("proxy.config.ssl.client.private_key.filename");
+  sslClientUpdate.reset(new ConfigUpdateHandler<SSLClientCoordinator>());
```
Comment (maskit) [inline iocore/net/SSLConfig.cc:443]: Why don't you have this line in `SSLClientCoordinator::startup()` ?

Comment (shinrich) [reply iocore/net/SSLConfig.cc:443]: I didn't create a SSLClientCoordinator::startup().  Only introduced the class for the reconfigure case.  We hadn't seem ordering problems on a process start.

Probably to be consistent, we could insert SSLClientCoordinator in the startup path as well.

Comment (maskit) [reply iocore/net/SSLConfig.cc:443]: Yeah, but not only for consistency. If `SSLClientCoordinator` is responsible for making sure the two configurations are loaded in the right order, I think it should take all the responsibility include the startup path.

File: iocore/net/SSLConfig.cc [hunk 2]
```diff
@@ -43,6 +43,7 @@
 #include "P_Net.h"
 #include "P_SSLClientUtils.h"
 #include "P_SSLCertLookup.h"
+#include "P_SSLSNI.h"
```
Comment (maskit) [inline iocore/net/SSLConfig.cc:46]: I assume this is for `SNIConfig::reconfigure();`. In this way, SSLConfig is going to be another SSLUtil. Let's separate out SSLClientCoordinator.

**Comment by maskit** [comment]
There are places that only call SSLConfig::startup(), and those would cause SNIConfig::reconfigure() without its startup(),

**Comment by shinrich** [comment]
pushed new commit to address @maskit's issues.

**Comment by shinrich** [comment]
I think the last clang-analyzer failed due to a problem with the server, e.g. out of disk or something. Seeing this at the end of the job
```
Error in tempfile() using template /CA/clang-analyzer/github/7313/2020-11-23-123120-12975-1/temp_buf_XXXXXX: Parent directory (/CA/clang-analyzer/github/7313/2020-11-23-123120-12975-1/) does not exist at /opt/llvm/bin/../libexec/ccc-analyzer line 34.
make[2]: *** [healthchecks/healthchecks.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  /opt/llvm/bin/../libexec/ccc-analyzer -DHAVE_CONFIG_H -I. -I../include -Dlinux -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/var/jenkins/workspace/clang-analyzer-github/src/proxy/api -I/var/jenkins/workspace/clang-analyzer-github/src/proxy/api -I/var/jenkins/workspace/clang-analyzer-github/src/include/cppapi/include -I/var/jenkins/workspace/clang-analyzer-github/src/lib/cppapi/include -I/var/jenkins/workspace/clang-analyzer-github/src/include -I/var/jenkins/workspace/clang-analyzer-github/src/lib -I/var/jenkins/workspace/clang-analyzer-github/src/lib/yamlcpp/include -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=gnu99 -g -pipe -Wall -Wno-deprecated-declarations -Qunused-arguments -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -O3 -fno-strict-aliasing -mcx16 -MT libloader/libloader.lo -MD -MP -MF libloader/.deps/libloader.Tpo -c libloader/libloader.c  -fPIC -DPIC -o libloader/.libs/libloader.o
Error in tempfile() using template /CA/clang-analyzer/github/7313/2020-11-23-123120-12975-1/temp_buf_XXXXXX: Parent directory (/CA/clang-analyzer/github/7313/2020-11-23-123120-12975-1/) does not exist at /opt/llvm/bin/../libexec/ccc-analyzer line 34.
make[2]: *** [libloader/libloader.lo] Error 1
mv -f compress/.deps/compress_compress_la-misc.Tpo compress/.deps/compress_compress_la-misc.Plo
mv -f compress/.deps/compress_compress_la-configuration.Tpo compress/.deps/compress_compress_la-configuration.Plo
make[2]: Leaving directory `/var/jenkins/workspace/clang-analyzer-github/src/plugins'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/var/jenkins/workspace/clang-analyzer-github/src/plugins'
make: *** [all-recursive] Error 1
scan-build: No bugs found.
```

**Comment by shinrich** [comment]
[approve ci clang-analyzer]

**Review by maskit — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**